### PR TITLE
JWT-powered Delegated Authentication

### PR DESCRIPTION
**Pull request by reyattrestle**
## Description

This PR adds a new passport strategy based on `passport-jwt` that will use JSON Web Tokens to delegate authentication to a third-party app. This essentially is a stripped down version of what Auth0 provides, but meant to use where both systems are maintained by the same group.

Note about length of this PR: it includes docs and `package.json` changes, which makes it go over the 300 lines limit.

# Checklist:

- [X] I have manually tested my changes on desktop and mobile
- [X] The test suite passes locally with my changes
- [ ] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [ ] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#submitting-your-pull-request), or has a documented reason in the description why it’s longer
- [X] I have made any necessary changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -50,3 +50,10 @@ DST_REFERENCE_TIMEZONE='US/Eastern'
 PASSPORT_STRATEGY=local
 TEXTER_SIDEBOXES=celebration-gif,default-dynamicassignment,default-releasecontacts,contact-reference,tag-contact,freshworks-widget,default-editinitial,take-conversations,hide-media,texter-feedback
 OWNER_CONFIGURABLE=ALL
+
+
+#PASSPORT_STRATEGY=token
+#TOKEN_AUTH_URL=
+#TOKEN_AUTH_SHARED_SECRET=secret
+#TOKEN_AUTH_ISSUER=issuer
+#TOKEN_AUTH_AUDIENCE=audience
```
Comment (crayolakat) [inline .env.example]: Can you move this example configuration to the `docs/HOWTO_INTEGRATE_TOKEN_AUTH.md` file?

File: Dockerfile [hunk 2]
```diff
@@ -21,6 +21,7 @@ RUN yarn install --ignore-scripts --non-interactive --frozen-lockfile && \
 FROM ${RUNTIME_IMAGE}
 WORKDIR /spoke
 COPY --from=builder /spoke/build build
+COPY --from=builder /spoke/migrations migrations
```
Comment (crayolakat) [inline Dockerfile]: Why is this needed?

File: package.json [hunk 3]
```diff
@@ -197,6 +199,7 @@
     "regenerator-runtime": "^0.10.5",
     "selenium-webdriver": "^3.6.0",
     "sqlite3": "^4.1.1",
+    "supertest": "^6.2.3",
```
Comment (crayolakat) [inline package.json:202]: What gains are you getting from using `supertest` vs the current libraries present in the codebase?

**Review by crayolakat — CHANGES_REQUESTED** [review]
Hey Rey, thanks for this PR! Very well written and useful new passport strategy.

I just had a few small questions, and a request to move the example configuration out of the .env.example file and into the `docs/` folder.

File: .env.example (hunk 1 above)
Comment (reyattrestle) [reply .env.example]: Sure thing!

File: Dockerfile (hunk 2 above)
Comment (reyattrestle) [reply Dockerfile]: It is not! I was just confirming, but because of how I have this sequenced in my docker-compose file, the postgres database is creating while Spoke is spinning up, which I was having trouble with because even with `SUPPRESS_DATABASE_AUTOCREATE` the migrations were still being called.

I just read 10 lines-ish below and saw that I can just have the same effect by setting `SUPPRESS_MIGRATIONS` and `SUPPRESS_SEED_CALLS` to truthy values.

I'll take this out.

File: package.json (hunk 3 above)
Comment (reyattrestle) [reply package.json:202]: The main feature I am leveraging from supertest is the ability to test out http requests in process without having to set up a server. I am using this feature in a test to allow me to set up the configuration params of the JWT issuer to be something unique to the test, and supertest allows me to assert on the response.

Basically these two assertions here:
```js
       // Login callback will send user to the appropriate page with the right token
        await request(app)
          .get(`/login-callback?nextUrl=/admin/${orgId}/campaigns`)
          .set("Authorization", `Bearer ${signedToken}`)
          .expect(302)
          .expect("Location", `/admin/${orgId}/campaigns`);

        // Bad tokens get rejected
        await request(app)
          .get(`/login-callback`)
          .set("Authorization", `Bearer ${invalidToken}`)
          .expect(401);
 ```

File: .env.example (hunk 1 above)
Comment (reyattrestle) [reply .env.example]: As a matter of fact, it's already there, so I'll just remove from the `.env.example` file

**Review by crayolakat — APPROVED** [review]
